pkg/instancegroups/scaleup.go

```diff
@@ -0,0 +1,196 @@
+	GroupUpdate *RollingUpdateInstanceGroup
+}
+
+// NewScaleUpRollout creates and returns a scale provider
+func NewScaleUpRollout(update *RollingUpdateInstanceGroup) Rollout {
+	return &scaleProvider{GroupUpdate: update}
+}
+
+// RollingUpdate performs a duplicate rolling update of an instance group
+func (p *scaleProvider) RollingUpdate(ctx context.Context, list *api.InstanceGroupList) error {
+	group := p.GroupUpdate.CloudGroup.InstanceGroup
+	fullname := p.GroupUpdate.Name
+	name := group.Name
+	strategy := group.Spec.Strategy
+	update := p.GroupUpdate.Update
+
+	// @step: increase the size of the instancegroup by x percent
+	var newMinSize, newMaxSize, oldMinSize, oldMaxSize int32
+	total := int32(p.GroupUpdate.CloudGroup.Size())
+	oldMinSize = int32(p.GroupUpdate.CloudGroup.MinSize)
+	oldMaxSize = int32(p.GroupUpdate.CloudGroup.MaxSize)
+	newMinSize = int32(p.GroupUpdate.CloudGroup.Size() * 2)
+	newMaxSize = int32(p.GroupUpdate.CloudGroup.MaxSize)
+	if newMinSize > newMaxSize {
+		newMaxSize = newMinSize
+	}
+	update.Infof("strategy: %s, adjusting size setting on instancegroup: %s, min/max (%d/%d)",
+		strategy.Name, name, newMinSize, newMaxSize)
+
+	update.Infof("updating the configuration with adjusted instancegroup settings for: %s", name)
+
+	// @step: grab the configuration from source
+	config, err := update.Clientset.InstanceGroupsFor(update.Cluster).Get(name, v1.GetOptions{})
+	if err != nil {
+		return update.Errorf("unable to retrieve configuration for instancegroup: %s, error: %v", name, err)
+	}
+	config.Spec.MinSize = &newMinSize
+	config.Spec.MaxSize = &newMaxSize
+
+	// @step: attempt to adjust the size of the instancegroup
+	if _, err := update.Clientset.InstanceGroupsFor(update.Cluster).Update(config); err != nil {
+		return update.Errorf("unable to update instancegroup: %s configuration, error: %v", name, err)
+	}
+
+	// @step: we need to update the termination policy to ensure if there is an error the new nodes are the
+	// are the ones which get kicked out first
+	update.Infof("updating termination policy on the instancegroup: %s to newest first", name)
+	if err := update.Cloud.SetTerminationPolicy(update.Cluster, fullname, cloudinstances.TerminateNewestPolicy); err != nil {
+		return update.Errorf("unable to update termination policy on instancegroup: %s, error: %v", name, err)
+	}
+
+	// @step: wrap the functionality for a rollback
+	result := func() error {
+		// @step: attempt to update the cluster
+		if err := update.UpdateCluster(ctx); err != nil {
+			return fmt.Errorf("unable to update the cluster, error: %v", err)
+		}
+		update.Infof("successfully update the configuration for instancegroup: %s", name)
+
+		// @step: we need to wait for a node interval and then attempt to validate cluster
+		update.Infof("waiting to %s while the instancegroup: %s to rescale", update.ScaleTimeout, name)
+		if err := p.GroupUpdate.WaitForGroupSize(ctx, name, int(newMinSize), update.ScaleTimeout); err != nil {
```

Comment:
@justinsb ... really dislike this, I'd prefer to wait for the nodes to appear as 'Ready' / 'NotReady' before moving on here .. I'm relying on group size here, but I couldn't see an easy way of filtering the nodes for a instancegroup. All I could think of was adding to the fi.Cloud to grab the instanceId and do something simpler to how the [GetCloudGroup](https://github.com/gambol99/kops/blob/rollout/upup/pkg/fi/cloudup/awsup/aws_cloud.go#L448-L479)